src/features/metadata-based-view/MetadataBasedItemList.js

```diff
@@ -0,0 +1,135 @@
+/**
+ * @flow
+ * @file Item list component
+ * @author Box
+ */
+
+import React from 'react';
+import classNames from 'classnames';
+import { injectIntl } from 'react-intl';
+import MultiGrid from 'react-virtualized/dist/es/MultiGrid/MultiGrid';
+import AutoSizer from 'react-virtualized/dist/es/AutoSizer';
+import { getFileExtension } from '../../utils/file';
+import FileIcon from '../../icons/file-icon';
+import '../../elements/content-explorer/ItemList.scss';
+import './MetadataBasedItemList.scss';
+
+type Props = {
+    currentCollection: Collection,
+    metadataColumnsToShow: Array<string>,
+};
+
+type CellRendererArgs = {
+    columnIndex: number,
+    key: string,
+    rowIndex: number,
+    style: Object,
+};
+
+const FILENAME_COLUMN_WIDTH = 350;
+const METADATA_MIN_COLUMN_WIDTH = 250;
+
+let itemCollection;
+let metadataColumns;
+
+const getFileIcon = (filename: string = '') => {
+    const ext = getFileExtension(filename);
+
+    return ext ? <FileIcon dimension={32} extension={ext} /> : <FileIcon dimension={32} />;
+};
+
+const getColumnWidth = width => {
+    const availableWidth = width - FILENAME_COLUMN_WIDTH;
+    // Maintain min column width, else occupy the rest of the space equally
+    const mdColumnWidth = Math.max(availableWidth / metadataColumns.length, METADATA_MIN_COLUMN_WIDTH);
+
+    return ({ index }) => {
+        return index === 0 ? 350 : mdColumnWidth;
+    };
+};
+
+const getGridHeader = columnIndex => {
+    if (columnIndex === 0) {
+        // fixed column filename header
+        return <span>Name</span>;
```

Comment:
Are these plain spans needed? Can we make the markup have more semantic meaning?